js/piwik.js

```diff
@@ -2502,12 +2505,16 @@ if (typeof window.Matomo !== 'object') {
              * or before being sent as GET parameters
              */
             function purify(url) {
-                var targetPattern;
+                var targetPattern, i;
 
                 // we need to remove this parameter here, they wouldn't be removed in Matomo tracker otherwise eg
                 // for outlinks or referrers
                 url = removeUrlParameter(url, configVisitorIdUrlParameter);
 
+                for (i = 0; i < configExcludedQueryParams.length; i++) {
+                    url = removeUrlParameter(url, configExcludedQueryParams[i]);
```

Comment:
btw does `removeUrlParameter` remove parameters case insensitive? If someone was to configure `phpsessionid` for example I wonder if it should also remove `PHPSESSIONID`

Reply:
At the moment it's case sensitive, easy enough to change removeUrlParameter() to be case insensitive. It's an edge case, but that could produce unwanted effects, eg. if a site had multiple query parameters with the same name in a different case used for different things. ?pid=123&pID=abc Then both would be removed.

Reply:
let's keep it for now as is and not change anything 👍